File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/person/country-region/country-region-lookup-model.ts

export interface ICountryRegionLookupModel {
  countryRegionCode: string;
  name: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class CountryRegionLookupModel implements ICountryRegionLookupModel {
  countryRegionCode: string;
  name: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<CountryRegionLookupModel>) {
    if (!!init) {
      Object.assign<CountryRegionLookupModel, Partial<CountryRegionLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Person/CountryRegion/Commands/Queries/GetCountryRegions/CountryRegionLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Person;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Sales.CountryRegionCurrency.Queries.GetCountryRegionCurrencies;
namespace AdventureWorks.Application.DataEngine.Person.CountryRegion.Queries.GetCountryRegions
{
    public partial class CountryRegionLookupModel : IHaveCustomMapping
    {
        public string CountryRegionCode { get; set; }
        public string Name { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Person.CountryRegion, CountryRegionLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<CountryRegionLookupModel, BaseCountryRegion>().IgnoreMissingDestinationMembers();
        }
    }
}
